HPXMLtoOpenStudio/resources/hpxml.rb

```diff
@@ -498,8 +498,10 @@ def has_fuel(fuel, hpxml_doc = nil)
      'HeatPumpFuel',
      'BackupSystemFuel',
      'FuelType',
-     'IntegratedHeatingSystemFuel'].each do |fuel_name|
-      if XMLHelper.has_element(hpxml_doc, "//#{fuel_name}[text() = '#{fuel}']")
+     'IntegratedHeatingSystemFuel',
+     'Heater/Type'].each do |fuel_name|
+      if XMLHelper.has_element(hpxml_doc, "//#{fuel_name}[text() = '#{fuel}']") ||
+         (XMLHelper.has_element(hpxml_doc, "//#{fuel_name}[text() = '#{HPXML::HeaterTypeGas}']") && (fuel == HPXML::FuelTypeNaturalGas))
```

Comment:
This is an expensive operation (the xpath expression looks through every node (`//...`) of the HPXML file) and now you are doubling the number of searches for each item. You should be able to easily avoid that.